test/lifecycle_test.dart

```diff
@@ -134,6 +131,42 @@ main() {
         isComponent2: true,
       );
 
+      group('propTypes', () {
+        bool consoleErrorCalled;
+        var consoleErrorMessage;
+        JsFunction originalConsoleError;
+        DivElement mountNode;
+
+        setUp(() {
+          consoleErrorCalled = false;
+          consoleErrorMessage = null;
+          mountNode = new DivElement();
+
+          originalConsoleError = context['console']['error'];
+          context['console']['error'] = new JsFunction.withThis((self, message, arg1, arg2, arg3) {
+            consoleErrorCalled = true;
+            consoleErrorMessage = message;
+
+            originalConsoleError.apply([message], thisArg: self);
+          });
+        });
+
+        tearDown(() {
+          context['console']['error'] = originalConsoleError;
+        });
+
+        test('fails validation with incorrect prop type', () {
+          react_dom.render(components2.PropTypesTest({'intProp': 'test'}), mountNode);
+          expect(consoleErrorCalled, isTrue, reason: 'should have outputted a warning');
+          expect(consoleErrorMessage, contains('intProp should be int'));
+        });
```

Comment:
Could we have an additional test here:
```dart
test('passes through all of the expected arguments to the prop validator', () { ... }
```
just to get coverage on this, maybe just dump all of those vars in an error message.

Reply:
`test('passes through all of the expected arguments to the prop validator', () { ... });`
FTFY

Reply:
https://github.com/cleandart/react-dart/pull/192/files#diff-c00a02aa016b25a86fb30a2093f935a1R171-R190 lulz